handler.go

```diff
@@ -94,19 +94,49 @@ func isIPv6(addr string) bool {
 	return ip != nil && ip.To4() == nil
 }
 
+// A responseWriterWithFirstWrite is an http.Responsewrite with the ability to
+// tracks the time when the first response write happened.
+type responseWriterWithFirstWrite interface {
+	http.ResponseWriter
+	didWrite()
+	firstWrite() time.Time
+}
+
 // A timedResponseWriter tracks the time when the first response write
 // happened.
 type timedResponseWriter struct {
-	firstWrite time.Time
+	firstWriteTime time.Time
```

Comment:
this doesn't need, nor warrant a rename

Reply:
It has been a long time since I made - for some reason I decided that an interface is needed. Will check why and come back to you.